File: eFood.Model/Lokacija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eFood.Model
{
    public class Lokacija
    {
        public int LokacijaId { get; set; }      

        public int KorisnikId { get; set; }    

        public double Latitude { get; set; }     
        public double Longitude { get; set; }    
        public DateTime Vrijeme { get; set; }
        public virtual Korisnik Korisnik { get; set; }

    }
}

File: UI/efood_mobile/lib/models/lokacija.dart
class Lokacija {
  int? lokacijaId;
  int? korisnikId;
  double latitude;
  double longitude;
  DateTime? datum;
  dynamic korisnik;

  Lokacija({
    this.lokacijaId,
    this.korisnikId,
    required this.latitude,
    required this.longitude,
    this.datum,
    this.korisnik,
  });

  factory Lokacija.fromJson(Map<String, dynamic> json) => Lokacija(
        lokacijaId: json['lokacijaId'],
        korisnikId: json['korisnikId'],
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
        datum: json['vrijeme'] != null ? DateTime.parse(json['vrijeme']) : null,
        korisnik: json['korisnik'],
      );

  /// Partial JSON (samo latitude/longitude)
  factory Lokacija.fromPartialJson(Map<String, dynamic> json) => Lokacija(
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
      );
}
